alibi_detect/saving/tests/test_saving.py

```diff
@@ -202,7 +202,10 @@ def nlp_embedding_and_tokenizer(model_name, max_len, uae, backend):
     backend = 'tf' if backend == 'tensorflow' else 'pt'
 
     # Load tokenizer
-    tokenizer = AutoTokenizer.from_pretrained(model_name)
+    try:
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+    except OSError:
```

Comment:
Is `OSError` the only one that can happen? Seems a bit weird here, is it because the network call was done somewhere else and failed silently and the loading is attempted from the locally downloaded file (which doesn't exist) or is `from_pretrained` performing the network call but `hugginface` isn't raising network errors up the stack?

Reply:
This is the full error traceback:

https://github.com/SeldonIO/alibi-detect/runs/7521461525

I think the `huggingface` strategy must be what you described i.e. if a `HTTPError` then they try to load from cache or a local directory. I tried passing an incorrect `model_name` to `from_pretrained` and using `HTTPError` or `ValueError` in `except`, but then I get the following:

`OSError: We couldn't connect to 'https://huggingface.co/' to load this model, couldn't find it in the cached files and it looks like bert-base-cased is not the path to a directory containing a config.json file.`

Seems to be like the other errors are not raised up the stack as you say?

Reply:
Maybe the solution might just be to do `except (OSError, HTTPError)`?

Reply:
Yea probably, in case they decide to change the error type at some point.

Reply:
Done!